server/src/main/java/io/littlehorse/common/model/metadatacommand/subcommand/PutWfSpecRequestModel.java

```diff
@@ -97,10 +105,32 @@ public WfSpec process(MetadataCommandExecution executionContext) {
         Optional<WfSpecModel> optWfSpec = oldVersion == null ? Optional.empty() : Optional.of(oldVersion);
 
         spec.validateAndMaybeBumpVersion(optWfSpec);
+        if (optWfSpec.isPresent() && WfSpecUtil.equals(spec, oldVersion)) {
+            return optWfSpec.get().toProto().build();
+        }
+
+        verifyUpdateType(allowedUpdateType, spec, optWfSpec);
         metadataManager.put(spec);
         return spec.toProto().build();
     }
 
+    private void verifyUpdateType(AllowedUpdateType updateType, WfSpecModel spec, Optional<WfSpecModel> oldSpec) {
+        switch (updateType) {
```

Comment:
I'd like to have a checkstyle rule which throws an error if any enum types are missing from a switch. Could we please address that here?

Reply:
You mean this?

![image](https://github.com/littlehorse-enterprises/littlehorse/assets/1911813/2dcad35d-3f4d-420e-b9ec-0b39fca343da)

Reply:
Yes, that. Thanks!